pkg/apis/workflow.go

```diff
@@ -102,3 +102,187 @@ func CreateWorkflow(requestData model.ChaosWorkFlowRequest, cred types.Credentia
+			return WorkflowListData{}, err
+		}
+
+		if len(workflowList.Errors) > 0 {
+			return WorkflowListData{}, errors.New(workflowList.Errors[0].Message)
+		}
+
+		return workflowList, nil
+	} else {
+		return WorkflowListData{}, err
+	}
+}
+
+type WorkflowRunsListData struct {
+	Errors []struct {
+		Message string   `json:"message"`
+		Path    []string `json:"path"`
+	} `json:"errors"`
+	Data WorkflowRunsList `json:"data"`
+}
+
+type WorkflowRunsList struct {
+	ListWorkflowRunsDetails model.ListWorkflowRunsResponse `json:"listWorkflowRuns"`
+}
+
+type GetChaosWorkFlowRunsGraphQLRequest struct {
+	Query     string `json:"query"`
+	Variables struct {
+		GetChaosWorkFlowRunsRequest model.ListWorkflowRunsRequest `json:"request"`
+	} `json:"variables"`
+}
+
+// GetWorkflowRunsList sends GraphQL API request for fetching a list of workflow runs.
+func GetWorkflowRunsList(in model.ListWorkflowRunsRequest, cred types.Credentials) (WorkflowRunsListData, error) {
+
+	var gqlReq GetChaosWorkFlowRunsGraphQLRequest
+	var err error
+
+	gqlReq.Query = `query listWorkflowRuns($request: ListWorkflowRunsRequest!) {
+                      listWorkflowRuns(request: $request) {
+                        totalNoOfWorkflowRuns
+                        workflowRuns {
+                          workflowRunID
+                          workflowID
+                          clusterName
+                          workflowName
+                          projectID
+                          clusterID
+                          clusterType
+                          isRemoved
+                          lastUpdated
+                          phase
+                          resiliencyScore
+                          experimentsPassed
+                          experimentsFailed
+                          experimentsAwaited
+                          experimentsStopped
+                          experimentsNa
+                          totalExperiments
+                          executedBy
+                        }
+                      }
+                    }`
+	gqlReq.Variables.GetChaosWorkFlowRunsRequest = in
+
+	query, err := json.Marshal(gqlReq)
+	if err != nil {
+		return WorkflowRunsListData{}, err
+	}
+
+	resp, err := SendRequest(
+		SendRequestParams{
+			Endpoint: cred.Endpoint + utils.GQLAPIPath,
+			Token:    cred.Token,
+		},
+		query,
+		string(types.Post),
+	)
+	if err != nil {
+		return WorkflowRunsListData{}, err
+	}
+
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	defer resp.Body.Close()
+	if err != nil {
+		return WorkflowRunsListData{}, err
+	}
+
+	if resp.StatusCode == http.StatusOK {
+		var workflowRunsList WorkflowRunsListData
+		err = json.Unmarshal(bodyBytes, &workflowRunsList)
+		if err != nil {
+			return WorkflowRunsListData{}, err
+		}
+
+		if len(workflowRunsList.Errors) > 0 {
+			return WorkflowRunsListData{}, errors.New(workflowRunsList.Errors[0].Message)
+		}
+
+		return workflowRunsList, nil
+	} else {
+		return WorkflowRunsListData{}, err
```

Comment:
This block will be executed if resp.StatusCode != http.StatusOK. In that case else block will return WorkflowRunsListData{}, nil since error will always be handled on line 269. So instead we can return custom error message.